cpu/atmega2560/periph/gpio.c

```diff
@@ -102,6 +102,24 @@ int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
     else {
         _SFR_MEM8(_ddr_addr(pin)) &= ~(1 << _pin_num(pin));
         res = bit_is_clear(_SFR_MEM8(_ddr_addr(pin)), _pin_num(pin));
+
+        if (res == 0) {
```

Comment:
Ah, yes I see. I'm sorry this macros confused me a little :-).
I test it with RIOT/tests/periph_gpio, maybe the test is buggy. I'll look at it later.